src/Subscription.ts

```diff
@@ -168,37 +189,25 @@ export class Subscription implements ISubscription {
    * @return {void}
    */
   remove(subscription: Subscription): void {
-
-    // HACK: This might be redundant because of the logic in `add()`
-    if (subscription == null   || (
-        subscription === this) || (
-        subscription === Subscription.EMPTY)) {
-      return;
-    }
-
-    const subscriptions = (<any> this)._subscriptions;
-
+    const subscriptions = this._subscriptions;
     if (subscriptions) {
       const subscriptionIndex = subscriptions.indexOf(subscription);
       if (subscriptionIndex !== -1) {
         subscriptions.splice(subscriptionIndex, 1);
       }
     }
   }
-}
-
-export class ChildSubscription extends Subscription {
-  constructor(private _innerSub: ISubscription, private _parent: Subscription) {
-    super();
-  }
 
-  _unsubscribe() {
-    const { _innerSub, _parent } = this;
-    _parent.remove(this);
-    _innerSub.unsubscribe();
+  private _addParent(parent: Subscription) {
+    let { _parent, _parents } = this;
+    if (!_parent) {
+      this._parent = parent;
+    } else if (_parents || (_parents = this._parents = [])) {
```

Comment:
Something like
```js
 } else {
  if (!_parents) {
    _parents = this._parents = [];
  }
  _parents.push(this);
 }
```

I get that these checks are annoying, perhaps there's a better place to set defaults.